src/mlpack/methods/pca/pca_main.cpp

```diff
@@ -57,26 +95,23 @@ int main(int argc, char** argv)
 
   // Get the options for running PCA.
   const size_t scale = CLI::HasParam("scale");
+  const double varToRetain = CLI::GetParam<double>("var_to_retain");
+  const string decompositionMethod = CLI::GetParam<string>(
+      "decomposition_method");
 
   // Perform PCA.
-  PCA p(scale);
-  Log::Info << "Performing PCA on dataset..." << endl;
-  double varRetained;
-  if (CLI::GetParam<double>("var_to_retain") != 0)
+  if (decompositionMethod == "exact")
   {
-    if (CLI::GetParam<int>("new_dimensionality") != 0)
-      Log::Warn << "New dimensionality (-d) ignored because --var_to_retain was"
-          << " specified." << endl;
-
-    varRetained = p.Apply(dataset, CLI::GetParam<double>("var_to_retain"));
+    RunPCA<ExactSVDPolicy>(dataset, newDimension, scale, varToRetain);
   }
-  else
+  else if(decompositionMethod == "randomized")
   {
-    varRetained = p.Apply(dataset, newDimension);
+    RunPCA<RandomizedSVDPolicy>(dataset, newDimension, scale, varToRetain);
+  }
+  else if(decompositionMethod == "quic")
+  {
+    RunPCA<QUICSVDPolicy>(dataset, newDimension, scale, varToRetain);
   }
```

Comment:
I think if a user picks an invalid decomposition policy, no error is issued? I am not sure if I am reading this right, but if so I think we should add an else to catch the error. :)